src/api-binder.coffee

```diff
@@ -0,0 +1,395 @@
+Promise = require 'bluebird'
+_ = require 'lodash'
+url = require 'url'
+TypedError = require 'typed-error'
+PlatformAPI = require 'pinejs-client'
+deviceRegister = require 'resin-register-device'
+express = require 'express'
+bodyParser = require 'body-parser'
+Lock = require 'rwlock'
+{ request, requestOpts } = require './lib/request'
+{ checkTruthy } = require './lib/validation'
+
+DuplicateUuidError = (err) ->
+	_.startsWith(err.message, '"uuid" must be unique')
+
+ExchangeKeyError = class ExchangeKeyError extends TypedError
+
+REPORT_SUCCESS_DELAY = 1000
+REPORT_RETRY_DELAY = 5000
+
+class APIBinderRouter
```

Comment:
I'll go with your second suggestion. The first one doesn't work because  of the way express defines the Router object https://github.com/expressjs/express/issues/2413